tsl/src/bgw_policy/job.c

```diff
@@ -839,6 +815,233 @@ job_execute(BgwJob *job)
 
 	if (transaction_started)
 		CommitTransactionCommand();
+}
+
+/*
+ * This code is based on:
```

Comment:
Would be good to outline the steps involved here and why it is needed.

Reply:
Sure